### PR TITLE
Add scoped mock documentation

### LINKED ISSUE #1114: Document a missing mockkObject() overload
### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

I expected to find information in the official documentation about this overload
```
inline fun mockkObject(vararg objects: Any, recordPrivateCalls: Boolean = false, block: () -> Unit)
```
which lets you create an object mock, execute the code inside the block against it, and then unmock it at the end of the block.

### Current Behavior

The official documentation is missing a section that describes the existence of this overload and its usage.

### PR DESCRIPTION
**Pull request by kshired**
resolve #1114

### CODE REVIEW

**Comment by kshired** [comment]
@Raibaz Can you give me some feedback of this PR?

File: README.md [hunk 1]
```diff
@@ -1155,6 +1155,24 @@ every { quit(1) } throws Exception("this is a test")
 * clear - deletes the internal state of objects associated with a mock, resulting in an empty object
 * unmock - re-assigns transformation of classes back to original state prior to mock
 
+### Scoped mock
```
Comment (Raibaz) [inline README.md:1158]: Should be "Scoped mocks"

File: README.md [hunk 2]
```diff
@@ -1155,6 +1155,24 @@ every { quit(1) } throws Exception("this is a test")
...
+
+Scoped mock is a mock that unmocks it after the block has been executed.
```
Comment (Raibaz) [inline README.md:1160]: "A scoped mock is a mock that automatically unmocks itself after the code block passed as a parameter has been executed".

File: README.md [hunk 3]
```diff
@@ -1155,6 +1155,24 @@ every { quit(1) } throws Exception("this is a test")
...
+You can use `mockkObject`, `mockkStatic` and `mockkConstructor` functions.
```
Comment (Raibaz) [inline README.md:1161]: "You can use the `mockkObject`..."

File: README.md [hunk 4]
```diff
@@ -1155,6 +1155,24 @@ every { quit(1) } throws Exception("this is a test")
...
+
+```kotlin
+object ObjBeingMocked {
+ fun add(a: Int, b: Int) = a + b
+}
+
+// ObjeBeingMocked will be reverted back after this scope.
```
Comment (Raibaz) [inline README.md:1168]: Should be "ObjBeingMocked will be unmocked after this scope"

**Review by Raibaz — CHANGES_REQUESTED** [review]

**Comment by kshired** [comment]
@Raibaz thaks!

File: README.md [hunk 5]
```diff
@@ -1155,6 +1155,24 @@ every { quit(1) } throws Exception("this is a test")
 * clear - deletes the internal state of objects associated with a mock, resulting in an empty object
 * unmock - re-assigns transformation of classes back to original state prior to mock
 
+### Scoped mocks
+
+A Scoped mock is a mock that automatically unmocks itself after the code block passed as a parameter has been executed
```
Comment (Raibaz) [inline README.md:1160]: Missing '.' at the end of the sentence.

File: README.md [hunk 6]
```diff
@@ -1155,6 +1155,24 @@ every { quit(1) } throws Exception("this is a test")
...
+You can use the `mockkObject`, `mockkStatic` and `mockkConstructor` functions.
+
+```kotlin
+object ObjBeingMocked {
+ fun add(a: Int, b: Int) = a + b
+}
+
+// Should be "ObjBeingMocked will be unmocked after this scope"
```
Comment (Raibaz) [inline README.md:1168]: I meant the comment should be just
`// ObjBeingMocked will be unmocked after this scope` :)

**Review by Raibaz — CHANGES_REQUESTED** [review]

**Review by Raibaz — APPROVED** [review]

**Pull request merged**